src/pages/workspace/categories/WorkspaceCategoriesPage.tsx

```diff
@@ -113,41 +133,119 @@ function WorkspaceCategoriesPage({policy, policyCategories, route}: WorkspaceCat
+        Navigation.navigate(ROUTES.WORKSPACE_CATEGORY_SETTINGS.getRoute(route.params.policyID, category.text));
     };
 
-    const navigateToCategorySettings = (category: PolicyForList) => {
-        Navigation.navigate(ROUTES.WORKSPACE_CATEGORY_SETTINGS.getRoute(route.params.policyID, category.text));
+    const navigateToCategoriesSettings = () => {
+        Navigation.navigate(ROUTES.WORKSPACE_CATEGORIES_SETTINGS.getRoute(route.params.policyID));
     };
 
     const navigateToCreateCategoryPage = () => {
         Navigation.navigate(ROUTES.WORKSPACE_CATEGORY_CREATE.getRoute(route.params.policyID));
     };
 
-    const isLoading = !isOffline && policyCategories === undefined;
+    const dismissError = (item: PolicyOption) => {
+        Policy.clearCategoryErrors(route.params.policyID, item.keyForList);
+    };
+
+    const selectedCategoriesArray = Object.keys(selectedCategories).filter((key) => selectedCategories[key]);
+
+    const getHeaderButtons = () => {
+        if (selectedCategoriesArray.length > 0) {
+            const isAllEnabled = selectedCategoriesArray.every((categoryName) => policyCategories?.[categoryName].enabled);
 
-    const headerButtons = (
-        <View style={[styles.w100, styles.flexRow, isSmallScreenWidth && styles.mb3]}>
-            {!PolicyUtils.hasAccountingConnections(policy) && (
+            const options: Array<DropdownOption<DeepValueOf<typeof CONST.POLICY.CATEGORIES_BULK_ACTION_TYPES>>> = [
+                {
+                    icon: Expensicons.Trashcan,
+                    text: translate('workspace.categories.deleteCategories'),
+                    value: CONST.POLICY.CATEGORIES_BULK_ACTION_TYPES.DELETE,
+                    onSelected: () => {
+                        setSelectedCategories({});
+                        deleteWorkspaceCategories(route.params.policyID, selectedCategoriesArray);
+                    },
+                },
+            ];
+
+            if (isAllEnabled) {
+                const categoriesToDisable = selectedCategoriesArray
+                    .filter((categoryName) => policyCategories?.[categoryName].enabled)
+                    .reduce<Record<string, {name: string; enabled: boolean}>>((acc, categoryName) => {
+                        acc[categoryName] = {
+                            name: categoryName,
+                            enabled: false,
+                        };
+                        return acc;
+                    }, {});
+
+                options.push({
+                    icon: Expensicons.DocumentSlash,
+                    text: translate('workspace.categories.disableCategories'),
+                    value: CONST.POLICY.CATEGORIES_BULK_ACTION_TYPES.DISABLE,
+                    onSelected: () => {
+                        setSelectedCategories({});
+                        setWorkspaceCategoryEnabled(route.params.policyID, categoriesToDisable);
+                    },
+                });
+            } else {
```

Comment:
I don't think this should be an if-else. If some categories are enabled and some are disabled, we want to show both options

Reply:
How should the functionality work if we select a mix of enabled and disabled categories? Should we have a feature that only enables the previously disabled categories, and another that only disables the previously enabled ones? What's your perspective on this?

Reply:
Yes, you can see how we're doing it for distance rates [here](https://github.com/Expensify/App/pull/37717/files#diff-bc07e9853f8d016b7882bd8bfc8c58153e0cee5b3b5f5f3ba0365ac20b59f8deR152-R183). If any selected category is enabled, we should add the `Disable` option, similar for disabled categories. We then filter the categories that don't need to change state before making the API call

Reply:
When configuring options, if a category's status changes (e.g., from enabled to disabled), should we still include categories that remain unchanged in our selection? For instance, when mixing enabled and disabled categories, if we decide to disable the previously enabled ones, should we deselect all categories or only those whose statuses have been altered?

Reply:
I think we should keep all of them selected, but the API call should filter out categories that are not changing state.